compiler/rustc_typeck/src/expr_use_visitor.rs

```diff
@@ -144,7 +144,13 @@ impl<'a, 'tcx> ExprUseVisitor<'a, 'tcx> {
         debug!("delegate_consume(place_with_id={:?})", place_with_id);
 
         let mode = copy_or_move(&self.mc, place_with_id);
-        self.delegate.consume(place_with_id, diag_expr_id, mode);
+
+        match mode {
+            ConsumeMode::Move => self.delegate.consume(place_with_id, diag_expr_id, mode),
+            ConsumeMode::Copy => {
+                self.delegate.borrow(place_with_id, diag_expr_id, ty::BorrowKind::ImmBorrow)
```

Comment:
Hmm, I had expected to make this change within the upvar delegate, and not the ExprUseVisitor. It seems potentially useful to some clients to know that this was a "by-value copy" -- I think there is other code in clippy that uses this, for example.

Reply:
Otherwise we should just remove the `ConsumeMode`